content/ui-patterns/saving.mdx

```diff
@@ -0,0 +1,163 @@
+
+Automatic saving applies changes as they are made.
+
+Automatic saving should be used when the user expects instant feedback from the change they made. Changes should only be saved automatically using[imperitive inputs](#imperitive-inputs) and for individual settings that are not changed via separate flows.
+
+If a field is saved automatically, it should be obvious whether or not it saved without using text or visual indicators. A visual indicator for a field that has been "saved" or has an error "error" could be mistaken as a validation status.
+
+![Screen recording of a user changing their theme in the 'Appearance' settings tab](https://user-images.githubusercontent.com/2313998/150868921-afc2b8b8-e39f-478f-b17d-398b790bda89.gif)
+
+
+### Imperitive inputs
+
+When user input is automatically saved, use the following inputs:
+- Toggle switches
+- Segmented controls
+- Single-select dropdowns
+
+These inputs are designed to reduce UI by giving instant feedback. The toggle switch behaves like a light switch: the light just switches on and off without you having to confirm each time you flip the switch. Segmented controls and single-select dropdowns behave like a soda machine: the soda comes out as soon as you press the buttons without you confirming which soda you wanted.
+
+To build an accessible and consistent app, do not automatically save declarative inputs.
+
+### Inputs that are problematic for autosaving:
+**Text inputs**
+- Users are expecting to set a value and then submit
+- Sensitive information could be unintentionally submitted. For example: a change password input
+
+**Checkbox groups**
+- It could be unclear whether or not their selections have been saved
+- Users may accidentally submit a selection that they clicked by accident
+
+**Radio button groups**
+Screenreader users can't read the options without selecting them. This could accidentally apply a selection the user didn't want.
+
+**Browser-native `<select>` dropdowns**
+On Windows, a select dropdown has a similar problem as a radio button group. 
+Options are selected when a user focuses the select and uses their arrow keys to read through each option
+
+## Feedback
+
+If the change may not be obvious in the UI within the user's viewport, provide feedback to let the user know that the change occurred successfully. If the change occured without a redirect or page refresh, you could use a toast message. If the change occured and the page refreshes or redirects, you could use a flash banner message.
+
+<img width="1787" alt="Screenshot of the user's activity feed page with a flash banner confirming that a repository has been deleted" src="https://user-images.githubusercontent.com/2313998/150868934-b96e6d9d-d022-433f-87ab-189baf1776fa.png">
+
+
+If an imperitive input did not automatically save, show an error message that explains why the change wasn't saved.
+
+{/*
+TODO: find an example of this, mock one up (if that's allowed), or exclude this visual example
```

Comment:
Any ideas where I could find an auto-saving field that shows an error message when the change encounters an error?

Reply:
Ignore this comment. I'm going to be replacing all screenshots with mockups that demonstrate the issue in isolation.